app/src/test/java/org/simple/clinic/bloodsugar/entry/BloodSugarEntryUiRendererTest.kt

```diff
@@ -0,0 +1,55 @@
+package org.simple.clinic.bloodsugar.entry
+
+import com.nhaarman.mockito_kotlin.mock
+import com.nhaarman.mockito_kotlin.verify
+import junitparams.JUnitParamsRunner
+import junitparams.Parameters
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.simple.clinic.bloodsugar.Fasting
+import org.simple.clinic.bloodsugar.PostPrandial
+import org.simple.clinic.bloodsugar.Random
+import org.simple.clinic.util.TestUserClock
+import org.threeten.bp.LocalDate
+import java.util.UUID
+
+@RunWith(JUnitParamsRunner::class)
+class BloodSugarEntryUiRendererTest {
+
+  private val patientUuid = UUID.fromString("79145baf-7a5c-4442-ab30-2da564a32944")
+
+  private val ui = mock<BloodSugarEntryUi>()
+  private val bloodSugarEntryUiRenderer = BloodSugarEntryUiRenderer(ui)
+
+  @Test
+  @Parameters(method = "params for creating model")
+  fun `it should render correct title based on the blood sugar measurment type`(
+      openAs: OpenAs
+  ) {
+    // given
+    val testUserClock = TestUserClock()
+    val year = LocalDate.now(testUserClock).year
+    val defaultModel = BloodSugarEntryModel.create(year, openAs)
+
+    // when
+    bloodSugarEntryUiRenderer.render(defaultModel)
+
+    // then
+    if (openAs is New) {
+      when (openAs.measurementType) {
```

Comment:
We should avoid branching in tests because it makes reading and maintaining them hard. Could we break this into three separate tests? They could be named

```kotlin
fun `it should render the title when creating new random blood sugar`() {}

fun `it should render the title when creating new post prandial blood sugar`() {}

fun `it should render the title when creating new fasting blood sugar`() {}
```

This will allow us to remove the parameterized tests as well. This is a tangential comment, but we've been moving away from parameterized tests since they are harder to maintain as well. We should always prefer separate discrete tests over parameterization or branching wherever possible.

Reply:
Done